editor.planx.uk/src/@planx/components/MapAndLabel/Public/Context.tsx

```diff
@@ -212,8 +213,9 @@ export const MapAndLabelProvider: React.FC<MapAndLabelProviderProps> = (
     resetErrors();
     removeFeatureFromForm(index);
     removeFeatureFromMap(index);
+
     // Set active index as highest tab after removal, so that when you "add" a new feature the tabs increment correctly
-    setActiveIndex((features && features.length - 2) || activeIndex - 1);
+    setActiveIndex((features && features.length - 2) || 0);
```

Comment:
If this is `- 1` rather than `- 2`, then there's a known bug:
- Add three features
- Remove feature 3 - active tab is tab 2
- Remove feature 2 - active tab is NOT tab 1 as expected

One to keep thinking on, still not completely satisfied with this solution !